public/app/features/explore/Graph.tsx

```diff
@@ -133,35 +133,29 @@ export class Graph extends PureComponent<GraphProps, GraphState> {
   }
 
   onPlotSelected = (event, ranges) => {
-    if (this.props.onChangeTime) {
+    const { onChangeTime, timeZone } = this.props;
+    if (onChangeTime) {
       const range = {
-        from: moment(ranges.xaxis.from),
-        to: moment(ranges.xaxis.to),
+        from: timeZone.isUtc ? moment.utc(ranges.xaxis.from) : moment(ranges.xaxis.from),
```

Comment:
Maybe Graph component should publish from/to in epoch ms instead of moment and the GraphContainer would be responsible for converting that to moment?

Reply:
yes, that could make things easier I think